bubble_sort.rb

```diff
@@ -0,0 +1,33 @@
+def bubble_sort(my_array)
+  loop do
+    my_array.map.with_index do |_num, i|
+      return my_array unless i != my_array.length - 1
+
+      if my_array[i] > my_array[i + 1]
+        my_array[i], my_array[i + 1] = my_array[i + 1], my_array[i]
+        break
+      end
+    end
+  end
+end
+ar = [4, 3, 78, 2, 98, 0, 2, 1]
+p bubble_sort(ar)
+
+def sort_by(array2)
+  loop do
+    array2.map.with_index do |_str, i|
+      return array2 unless i != array2.length - 1
+
+      if array2[i].length > array2[i + 1].length
+        array2[i], array2[i + 1] = array2[i + 1], array2[i]
+        break
+      end
+    end
+  end
+end
```

Comment:
- [ ] _"Now create a similar method called #bubble_sort_by which sorts an array by accepting a block. Remember to use yield inside your method definition to accomplish this."_ (Extracted from the pathwayright instructions).

For the second method, I don't see the use of the `yield`. This is a keyword that let's you extract the given block so you can use it inside your function. Using this you will be able to check if the user gave you a block, and if not you can throw the error.

Also, using the given block you will be able to sort from left to right or from right to left.

If this sounds confusing, please reach me through slack, I am more likely to respond between 9 am to 6 pm GTM -5.